src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -239,6 +239,8 @@ public FormValidation doCheckReturnStatus(@QueryParameter boolean returnStdout,
         private boolean watching; // serialized default is false
         /** Only used when {@link #watching}, if after {@link #WATCHING_RECURRENCE_PERIOD} comes around twice {@link #exited} has yet to be called. */
         private transient boolean awaitingAsynchExit;
+        /** The first throwable used to stop the task */
+        private transient volatile Throwable causeOfStoppage;
```

Comment:
I think this needs to be `volatile` for access from both `#stop` and `#exited`, but if so then I think `recurrencePeriod` needs to be volatile as well (`long`s might be effectively volatile on 64-bit machines, but that is not guaranteed by the [JLS](https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.7)).

Reply:
`stop` and `exited` could be called from unknown threads, so `volatile` seems appropriate.

`transient` is fine I think. There is a 10s grace period before forcible step termination, so that is the maximum we are dealing with. At worst, if there happens to be a restart at that moment, it would merely make this fix not take effect, thus yielding a `FAILURE`.